cli/ops.rs

```diff
@@ -1793,9 +1793,26 @@ fn op_run(
     c.env(entry.key().unwrap(), entry.value().unwrap());
   });
 
-  c.stdin(subprocess_stdio_map(inner.stdin()));
-  c.stdout(subprocess_stdio_map(inner.stdout()));
-  c.stderr(subprocess_stdio_map(inner.stderr()));
+  let stdin_rid = inner.stdin_rid();
+  if stdin_rid > 0 {
+    c.stdin(resources::get_file(stdin_rid)?);
```

Comment:
At least in unix, the stdio can be associated with many types of fds including pipes and sockets, not just files. Probably it doesn't work like that on Windows.

Some commentary here regarding this would be useful. Maybe a TODO to make this work with sockets. It seems like if you give a socket rid in your patch, it will result in a panic("bad rid")

Reply:
Added a TODO